compat/socket.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/socket.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+#include <map>
+
+static_assert(SOCKET_ERROR == -1);
+
+/**
+ * Squid socket code is written to handle POSIX errno codes.
+ * Set errno to the relevant POSIX or WSA code.
+ */ 
+static void
+SetErrnoFromWsaError()
+{
+    // POSIX codes which socket API users may care about
+    static const std::map<int, int> CodeMap = {
```

Comment:
Please use NoNewGlobals patterns for non-POD statics, especially those that, like this CodeMap, may be used after main() exits.

Reply:
What are you taking about? This is a local variable static, not a global.

Reply:
done in 50aebd7b5f